### ISSUE TITLE
The format list should be an ARIA menu and other a11y improvements

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#1879]
The format list with paragraph format and headings level should work like the DropDown component:

![screen shot 2017-07-13 at 09 23 23](https://user-images.githubusercontent.com/1682452/28159822-891b905c-67be-11e7-810b-a27c8992c977.png)

- should use `role="menu"` and  `role="menuitem"`
- should use arrows navigation
- all the items are `<span>` elements: with `role="menuitem"` their semantic would be OK but worth considering to use a natively interactive element (button?) to simplify the JS part
- not sure why it is using `aria-hidden="true"` on the items
- the toggle button needs a focus style, see for example the style used on the similar block switcher toggle button:

<img width="204" alt="screen shot 2017-07-13 at 11 34 17" src="https://user-images.githubusercontent.com/1682452/28160069-5b2f0844-67bf-11e7-9895-00af1b3df4e3.png">

Todo: consider to re-use the `DropdownMenu` component

**Comment by afercia** [comment]
I think this has been refactored and now uses the DropDown. However seems keyboard navigation wit arrows doesn't work as expected, needs to be tested again.

**Comment by afercia** [comment]
Seems to me at the moment this dropdown list of formats is used only in the "Classic Text" block.

**Comment by afercia** [comment]
See https://github.com/WordPress/gutenberg/pull/2989

**Comment by jeffpaul** [comment]
_This ticket was mentioned in [Slack](https://make.wordpress.org/chat/) in #core-editor by jeffpaul. [View the logs](https://wordpress.slack.com/archives/C02QB2JS7/p1519321344000545)._

**Comment by afercia** [comment]
I think this can be closed now. Since Jul 2017 many things have changed and blocks don't use this "dropdown" any longer. The only block that uses something similar is the "Classic" one, which is powered by TinyMCE.
